From: Harald Welte <laforge@gnumonks.org>
Date: Mon, 8 Jun 2009 10:19:16 +0000 (+0800)
Subject: CPUFREQ: Enable acpi-cpufreq driver for VIA/Centaur CPUs
X-Git-Url: http://git.gnumonks.org/cgi-bin/gitweb.cgi?p=linux-2.6-via.git;a=commitdiff_plain;h=03c869d3e851c585b30d0b1d115546406da2c692

CPUFREQ: Enable acpi-cpufreq driver for VIA/Centaur CPUs

The VIA/Centaur C7, C7-M and Nano CPU's all support ACPI based cpu p-states
using a MSR interface.  The Linux driver just never made use of it, since in
addition to the check for the EST flag it also checked if the vendor is Intel.

Signed-off-by: Harald Welte <HaraldWelte@viatech.com>
---

diff --git a/arch/x86/kernel/cpu/cpufreq/acpi-cpufreq.c b/arch/x86/kernel/cpu/cpufreq/acpi-cpufreq.c
index 54b6de2..36d4df2 100644
--- a/arch/x86/kernel/cpu/cpufreq/acpi-cpufreq.c
+++ b/arch/x86/kernel/cpu/cpufreq/acpi-cpufreq.c
@@ -90,7 +90,8 @@ static int check_est_cpu(unsigned int cpuid)
 {
 	struct cpuinfo_x86 *cpu = &cpu_data(cpuid);
 
-	if (cpu->x86_vendor != X86_VENDOR_INTEL ||
+	if ((cpu->x86_vendor != X86_VENDOR_INTEL &&
+	     cpu->x86_vendor != X86_VENDOR_CENTAUR) ||
 	    !cpu_has(cpu, X86_FEATURE_EST))
 		return 0;
 
